apps/3d_rec_framework/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ else(NOT VTK_FOUND)
 endif(NOT VTK_FOUND)
 
 # OpenNI found?
-if(NOT OPENNI_FOUND OR NOT BUILD_OPENNI)
+if(NOT OPENNI_FOUND OR NOT WITH_OPENNI)
```

Comment:
What about automatically coercing `WITH_OPENNI` to off if the package was not found? This will give the user some feedback, because `FOUND_` variables are not visible, and the user might thing that since `WITH_` is on, the corresponding modules will be build (when in fact they won't be).